core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -411,10 +439,14 @@ Builder scanMetrics(ScanMetrics newScanMetrics) {
       return this;
     }
 
-    DeleteFileIndex build() {
+    private Collection<DeleteFile> planFiles() {
+      if (deleteFiles != null) {
+        return deleteFiles;
+      }
+
       // read all of the matching delete manifests in parallel and accumulate the matching files in
       // a queue
-      Queue<ManifestEntry<DeleteFile>> deleteEntries = new ConcurrentLinkedQueue<>();
+      Queue<DeleteFile> files = new ConcurrentLinkedQueue<>();
```

Comment:
Why not `deleteFiles`?

Reply:
The name was clashing with a field but I will play around with it.